### DISCUSSION TITLE
Arch on M1 loses network connectivity after pacman -Syu

### DISCUSSION THREAD

**Discussion by zefram-cochrane** [utmapp/UTM#3392] [UTM Help (macOS)]
I wanted to update prior to putting EndeavourOS over arch so did

# pacman-key --init && pacman-key --populate archlinuxarm
# pacman -Syu

The update completes successfully and the VM still has connectivity. I then

# shutdown -h now

And restart. Now

# ip link

shows only the loopback on lo. Then

# dhcpcd eth0

tells me the eth0 interface is not found.

**Comment by pfremm** [comment]
This is because it's leveraging volumes outside the main disk image for the initial ramdisk and kernel.  When you do the upgrade the kernel modules are upgraded as well and the prior version is removed.  Pacman is updating the images in /boot but this VM isn't using that directory.

**Comment by eparreno** [comment]
I'm having the exact same issue. Any suggested workaround?

**Comment by pfremm** [comment]
A hack would be to scp /boot/Image.gz and /boot/initramfs-linux.img to another machine and copy them into the arch linux package in the Images directory.  While this would work it's probably going to be difficult to maintain each time a new kernel is released.

**Comment by zefram-cochrane** [comment]
Would it perhaps work to use pacman to downgrade the kernel and firmware packages after the update? I'm not hung up on the latest and greatest under the hood, just want to get a modern plasma environment installed without spending hours scratching my head in the guts of arch. 

In my ideal world I would be happier with Manjaro rather than Endeavouros , but expect it's foolhardy to go there given there's no pre-made manjaro in the UTM gallery ...

**Comment by pfremm** [comment]
That may work or ignoring that package from being upgraded is another option I guess for awhile.  At some point though you'll want an upgrade kernel.  You can also grab the Image.gz and initramfs-linux.img from http://os.archlinuxarm.org/os/ArchLinuxARM-aarch64-latest.tar.gz

**Comment by zefram-cochrane** [comment]
At some point, but by then I'd hope either the UTM gallery will have a better maintained option for a Plasma arch-based distro or - hopefully soon - there will be a decent native apple silicon linux. This way I need not hold my breath ;-)

**Comment by zefram-cochrane** [comment]
Well, progress. I can pacman -Syu this way and I get an apparently clean install on endeavouros plasma. But then lightdm crashes with a 727 so I'm stuck with just a tty ...

Googling makes that look like a pretty rarified issue so I'll try a reinstall using your hack above.

**Comment by MarinPostma** [comment]
@zefram-cochrane, I was in the same position as you, and I ended doing an installation myself, using images from here: https://www.reddit.com/r/archlinuxarm/comments/s0g2yy/archboot_iso_202201_released_for_aarch64/. I had issue with the installer, so I installed it manually. When installing grub, make sure to install it for the right architecture (omitting the `--target=x86_64-efi` in `grub-install` did the trick since it defaults to the right value).

After that, I get a working arch with working network, and an up to date  kernel. Hope that'll help you!

**Reply by Bicet** [reply]
can you maybe share your config?

I've some difficulties on booting the iso :( I mean I always get grub error :/
